elasticsearch-model/lib/elasticsearch/model/indexing.rb

```diff
@@ -153,7 +153,24 @@ def mapping(options={}, &block)
         #
         #     # => {:index=>{:number_of_shards=>1}}
         #
+        # You can specify a YAML file with settings
+        #
+        # @example Define index settings from YAML file
+        #
+        #     # config/elasticsearch/articles.yml:
+        #     #
+        #     # index:
+        #     #   number_of_shards: 1
+        #     #
+        #
+        #     Article.settings "config/elasticsearch/articles.yml"
+        #
+        #     Article.settings.to_hash
+        #
+        #     # => { "index" => { "number_of_shards" => 1 } }
+        #
         def settings(settings={}, &block)
+          settings = YAML.load_file(settings) if settings.is_a?(String)
```

Comment:
Sounds good, I'll make the changes and submit them. Thanks!

I'm also looking around for the best way to do it for mapping. In my apps I currently configure the whole index, including mappings, via a YAML file by patching the gem and adding a whole "configure_from_file" method that simply reads it and send it as the body of the index at creation. I don't know if you go that far for the gem. For now settings would help with the whole repeated analyzers thing.